### PR TITLE
Visual improvements & readme update + uwsgi friendly

### PR DESCRIPTION
**Pull request by fdev31**
- Better (?) CSS
- Few README lines about configuration (link to sources' pasttle.ini), probably should made in a separate pull request ...

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -106,6 +106,36 @@ starting the server, like this:
    export PASTTLECONF=/etc/pasttle/mypasttle.ini:development
    pasttle-server.py
 
+Available configuration options
+-------------------------------
+
+.. code:: ini
+
+    [main]
```
Comment (thekad) [inline README.rst]: Would be nice to list which are the default values for each option

File: src/pasttle/views/index.html [hunk 2]
```diff
@@ -11,7 +11,17 @@
...
         <h2>CLI Examples:</h2>
-        <div>
+        <script>
+        function show_examples() {
+            document.querySelector('#cli_examples').style.display = "block"
+            document.querySelector('#cli_helpers').style.display = "block"
+            document.querySelector('#cli_examples_holder').style.display = "none"
+        }
+        </script>
+        <div id="cli_examples_holder" onclick="show_examples()" style="cursor: pointer">
```
Comment (thekad) [inline src/pasttle/views/index.html]: This should also be underlined to show it's actually a link (I know it has the hand pointer style, but still.)

**Comment by fdev31** [comment]
Got it :)
Green proposals:

#2c7433

#5d742c

#2c742e

#4b7d4d

I'm not a colour expert, but they look a bit less aggressive, can you pick one? :)

I'll update my branch then, with the other comments applied too...

**Comment by thekad** [comment]
@fdev31 2c742e I think it looks best (in my personal opinion, of course) so if you wanna make the changes and update your branch, I can give it another look :smiley:

File: src/pasttle/util.py [hunk 3]
```diff
@@ -10,7 +10,13 @@
     import configparser
 import logging
 import os
-import StringIO
+
+try:
+    import StringIO as py2StringIO
```
Comment (thekad) [inline src/pasttle/util.py]: This would be better as:
```python
try:
  from StringIO import StringIO
except ImportError:
  from io import StringIO
```

Comment (fdev31) [inline src/pasttle/util.py]: I must have been very drunk when I did it... ^^

File: src/pasttle/views/index.html (hunk 2 above)
Comment (fdev31) [inline src/pasttle/views/index.html]: This is done but marked as pending...

**Comment by fdev31** [comment]
About the uwsgi integration, feel free to change it.
I personally really like the usage of .ini format for this "all-in-one" feature, having all the configuration in one place for one server, but I respect your taste!

**Comment by thekad** [comment]
@fdev31 oh I meant the uwsgi bash script, that should be in a contrib folder or just add as an example in README

File: src/pasttle/model.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ class Paste(Base):
 
     __tablename__ = 'paste'
 
-    id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True)
+    id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True, index=True, unique=True)
```
Comment (thekad) [inline src/pasttle/model.py]: I thought primary_key would set both of those other 2 flags automatically... I cannot find documentation supporting my assertion tho, I wonder if I dreamed it

Comment (fdev31) [inline src/pasttle/model.py]: Ok, about the fail, I checked and it's not functional test but something more personal, adapt the code to your own liking / constraints if it bugs you, I won't mind :)

**Comment by thekad** [comment]
1) After reading the sqlalchemy source code, I am pretty sure the model changes will have no effect: primary_key automatically turns on index and uniqueness on all RDBMS and, in fact, creating a database before and after the changes yields no difference in my testing
2) Even though you could argue the PEP8 tests are not functional, the fact that they belong to the test suite means the author (me in this case) considers them important enough to be respected, so let's do that :smiley:

With that being said, everything else looks correct, what do you say we remove the change in models (unless you got a test case where it actually makes a difference, I couldn't see one but I don't test all databases) and we can merge this?

**Comment by fdev31** [comment]
Ok, sorry but that starts to be boring... it's supposed to be help, not torture.
Can't you really add a newline after a merge?
Do we really have to debate explicit vs implicit?
Do we have to debate the advantages of an index? (I don't know how you tested, a quick benchmark showed 5% on a not-so-huge DB) Maybe you used some sophisticated database having "safe" defaults, but SQLite is the default choice.

That kind of thing is the reason why Github is not much used for contributions.
Maybe you can consider differences between contributing and maintaining/begin part of the regular developers.

So enjoy.
Oh, and my pretty sophisticated benchmark, since you can't write a decent one, without the caller bash script:

```python
import random
import time
from model import Paste, engine
from sqlalchemy.orm import Session
random.seed(1)

def create():
    session = Session(bind=engine)

    for n in range(100):
        session.add_all(Paste('<blah> nr %d-%d'%(n,j), 'text/plain') for j in range(1000))
        session.flush()
    session.commit()

def access():
    session = Session(bind=engine)
    maxi = 100 * 1000
    t0 = time.time()
    for n in range(10000):
        d = session.query(Paste).filter(Paste.id==random.randint(0, maxi)).all()
    print(time.time() - t0)

# create()
access()
```

Sorry, but you played with my nerves.

**Comment by thekad** [comment]
The initial changes in this PR were about changing the color and some other minimal UI tweaks. You kept adding changes to it afterwards, if you keep adding more and more changes, I am going to give more and more feedback, simple as that.
You are bringing a set of changes that you don't _actually_ need (pasttle-server can be configured pointing to a different template path which you can customize to your liking) yet you are getting mad because I am not merging those? that makes no sense to me.
I can add as many new lines I need to fix the test suite, the point is, it wasn't me who broke it in the first place. You will be asked to fix your broken stuff in any place you submit code to, regardless of it being open source or not.
I am not debating the usefulness of having indexes in a database, but since you didn't provide any metrics _before_ I asked and you have no plan on how to upgrade the schema from one version to the next (not even documentation) and even better those changes should be in a completely different pull request, I am not merging those.
And yeah, I don't have benchmarks, feel free to send another pull request with a bunch of them.

**Pull request merged**